### ISSUE TITLE
binding_ninja isn't JRuby compatible

### ISSUE THREAD

**Issue by mmcfarland-cavulus** [tomykaira/rspec-parameterized#49]
Hey y'all,

The new dependency on binding_ninja makes this library fail when compiling under JRuby. Here's the output:

```
current directory: /home/.rvm/gems/jruby-9.1.16.0/gems/binding_ninja-0.2.2/ext/binding_ninja
/home/.rvm/rubies/jruby-9.1.16.0/bin/jruby -r ./siteconf20181205-19971-1od2pas.rb extconf.rb
creating Makefile

current directory: /home/.rvm/gems/jruby-9.1.16.0/gems/binding_ninja-0.2.2/ext/binding_ninja
make "DESTDIR=" clean

current directory: /home/.rvm/gems/jruby-9.1.16.0/gems/binding_ninja-0.2.2/ext/binding_ninja
make "DESTDIR="
make: *** No rule to make target `/home/.rvm/rubies/jruby-9.1.16.0/lib/ruby/include/ruby/ruby.h', needed by `binding_ninja.o'.  Stop.

make failed, exit code 2
```

Looks like this library makes MRI-specific native extensions and can't be used under JRuby.

**Comment by mmcfarland-cavulus** [comment]
(repeating this comment form the PR)

There is some code that explicitly references the BindingNinja constant. I wonder if it would be worth refactoring that library to give it API-parity with binding_of_caller.

In the meantime, we've been using v0.4.0, and it's been working great.

**Comment by joker1007** [comment]
I have already released v0.4.2 that is supported JRuby.
